integration/terraform/ec2/win/main.tf

```diff
@@ -1,33 +1,71 @@
-resource "aws_instance" "integration-test" {
-  ami                    = data.aws_ami.latest.id
-  instance_type          = var.ec2_instance_type
-  key_name               = var.key_name
-  iam_instance_profile   = var.iam_instance_profile
-  vpc_security_group_ids = var.vpc_security_group_ids
-  get_password_data = true
+#Create a unique testing_id for each test
+resource "random_id" "testing_id" {
+  byte_length = 8
+}
+
+#####################################################################
+# Generate EC2 Key Pair for log in access to EC2
+#####################################################################
+
+resource "tls_private_key" "ssh_key" {
+  count     = var.ssh_key_name == "" ? 1 : 0
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "aws_ssh_key" {
+  count      = var.ssh_key_name == "" ? 1 : 0
+  key_name   = "ec2-key-pair-${random_id.testing_id.hex}"
+  public_key = tls_private_key.ssh_key[0].public_key_openssh
+}
+
+locals {
+  ssh_key_name        = var.ssh_key_name != "" ? var.ssh_key_name : aws_key_pair.aws_ssh_key[0].key_name
+  private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
+}
+
+#####################################################################
+# Generate EC2 Instance and execute test commands
+#####################################################################
+
+resource "aws_instance" "cwagent" {
+  ami                         = data.aws_ami.latest.id
+  instance_type               = var.ec2_instance_type
+  key_name                    = local.ssh_key_name
+  iam_instance_profile        = aws_iam_instance_profile.cwagent_instance_profile.name
+  vpc_security_group_ids      = [aws_security_group.ec2_security_group.id]
+  associate_public_ip_address = true
+  get_password_data           = true
+  tags = {
+    Name = "cwagent-integ-test-ec2-${var.test_name}-${random_id.testing_id.hex}"
+  }
+}
+
+resource "null_resource" "integration_test" {
+  depends_on = [aws_instance.cwagent]
   provisioner "remote-exec" {
-    # @TODO when @ZhenyuTan-amz adds windows tests add "make integration-test"
-    # @TODO add export for AWS region from tf vars to make sure runner can use AWS SDK
     inline = [
+      # @TODO when @ZhenyuTan-amz adds windows tests add "make integration-test"
+      # @TODO add export for AWS region from tf vars to make sure runner can use AWS SDK
```

Comment:
Why is it necessary to move the comment?

Reply:
No deep reason behind it. Forgot to changed it back 👍